website/docs/operators/update_operators/array-update-operators.md

```diff
@@ -0,0 +1,239 @@
+  }
+]
+```
+
+## $addToSet
+
+The `$addToSet` operator updates an array by adding a specified element to an array if the element does not already exist in the array.
+If the specified element exists in the array, the `$addToSet` operator will not modify the array.
+
+Insert the following documents into a `store` collection:
+
+```js
+db.store.insertMany([
+  { _id: 1, items: ["pens", "pencils"] },
+]);
+```
+
+**Example:** Use the `$addToSet` operator to update the array with non-existing elements.
+
+```js
+db.store.updateOne(
+  { _id: 1 },
+  { $addToSet: { items: "paper" } }
+);
+```
+
+The document is subsequently updated with the new element, as depicted below:
+
+```js
+[ { _id : 1, items : [ 'pens', 'pencils', 'paper' ] } ]
+```
+
+**Example:** Use the `$addToSet` operator to update the array with already existing elements.
+
+```js
+db.store.updateOne(
+  { _id: 1 },
+  { $addToSet: { items: "pens" } }
+);
+```
+
+Since the array already contains the element, there won't be any changes.
+
+```js
+[ {_id: 1, items: [ 'pens', 'pencils', 'paper' ] } ]
+```
+
+:::note
+The `$addToSet` is different from the `$push` operator which adds the element to the array either it exists or not.
+:::
+
+**Example:** Use the `$addToSet` operator for non-existing array fields.
+
+If the array field does not exist in the document, the `$addToSet` operator will create the field and add the element to the array.
+
+```js
+db.store.updateOne(
+  { _id: 1 },
+  { $addToSet: { colors: "red" } }
+);
+```
+
+The updated document looks like this:
+
+```js
+[
+  { _id: 1, items: [ 'pens', 'pencils', 'paper' ], colors: [ 'red' ] }
+]
+```
+
+## $pop
+
+With the `$pop` operator, you can update a document by removing the first or last element of an array.
+Assign a value of `-1` to remove the first element of an array, or `1` to remove the last element.
+
+Insert this document into a `products` collection:
+
+```js
+db.products.insertMany([
+    { _id: 1, items: [ "pens", "pencils", "paper", "erasers", "rulers" ] }
+]);
+```
+
+**Example:** Use the `$pop` operator to remove the first element of an array.
+
+```js
+db.products.updateOne(
+  { _id: 1 },
+  { $pop: { items: -1 } }
+);
+```
+
+The document is subsequently updated with the first element `pens` removed, as depicted below:
+
+```js
+[
+  {
+    _id: 1,
+    items: ['pencils', 'paper', 'erasers', 'rulers']
+  }
+]
+```
+
+To remove the last element of the array, assign `1` as the value for the `$pop` operator.
+
+```js
+db.products.updateOne(
+  { _id: 1 },
+  { $pop: { items: 1 } }
+);
+```
+
+The updated now looks like this:
+
+```js
+[
+  {
+    _id: 1,
+    items: ['pencils', 'paper', 'erasers']
+  }
+]
+```
+
+## $pullAll
+
+The `$pullAll` operator removes all the matching elements in a specified query from an array.
+
+Insert the following document into a `store` collection:
+
+```js
+db.store.insertMany([
+  { _id: 1, items: ["pens", "pencils", "paper", "erasers", "rulers"] },
```

Comment:
Should we have some duplicates (e.g. `paper` mentioned twice) to demonstrate how duplicate elements work with `$pullAll`?